### ISSUE TITLE
Update help documentation to reflect that `smooth_frames` isn't only applied for signals with multiple `samps_per_frame`

### ISSUE THREAD

**Issue by briangow** [MIT-LCP/wfdb-python#316]
Pull request #313 changes the behavior of `smooth_frames` such that it can be used to create a uniformly sampled array or a list of signals even when their `samps_per_frame = 1`.  The help documentation states that `smooth_frames` is only used when `samps_per_frame` is not 1.  This should be updated to reflect the new behavior.  Any updates made to the parameter descriptions regarding how the smoothing procedure works should also be reflected in the help documentation.

**Comment by bemoody** [comment]
Yeah, the documentation could be improved (and the API itself could be
improved.)

Are you just referring to the doc string for 'rdrecord', or do you see
other places in the documentation that are outdated or incomplete?

**Comment by briangow** [comment]
I was primarily thinking that the doc strings should be updated.  In this issue I mentioned that we could update the fact that the pending merge in #313 will change the behavior such that `smooth_frames` will also affect signals with `samps_per_frame=1`.  In #313 I also mentioned that we may consider expanding on the `smooth_frames` doc string to clarify how it works with respect to the signals `samps_per_frame` (i.e. `smooth_frames = True` will decimate the signal by its `samps_per_frame`).  

Beyond the doc strings it'd be great to have a somewhat high-level page on the PhysioNet site which clearly describes the WFDB formats.  I'm sure this is tricky to do while still including the necessary technical details and would take time away from higher priority tasks.  However, it'd be great to demystify WFDB a bit more so people are comfortable working with this format and using all of the nice tools that have been developed - my sense regarding some of the push back we get for requiring the WFDB format in project submissions is that there is confusion that could be alleviated with better documentation.  

I know I'm preaching to the choir but a nice overview of the toolboxes in addition to the format itself would also be welcome. I recently learned that a group that had used the WFDB toolboxes for some time didn't realize that signals could be loaded into temporary memory without explicitly downloading the signal first.  

Sorry for the rambling, only the first paragraph here is specific to this issue but it'd be great to hear your thoughts (@bemoody and others) regarding the documentation as a whole.
